src/libstore/build.cc

```diff
@@ -4412,14 +4416,16 @@ void SubstitutionGoal::tryNext()
     sub = subs.front();
     subs.pop_front();
 
-    if (sub->storeDir != worker.store.storeDir) {
-        tryNext();
-        return;
+    auto subPath = storePath;
+    if (ca) {
```

Comment:
Shouldn't we do `tryNext()` if `sub->storeDir != worker.store.storeDir && !ca`?

Reply:
@matthewbauer I think @edolstra is right? It will eventually try next after `queryPathInfo` fails, but we shouldn't wait for that guaranteed failure, right?

Reply:
this was addressed in 1f9cb06db23f357f06404969820be1d3d4118a18